### ISSUE TITLE
next/image preload not working as expected in AppDir. Affecting LCP and FP.  Order of preload is different than page-dir.

### ISSUE THREAD

**Issue by gauravsaini964** [vercel/next.js#52995]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:19 PDT 2023; root:xnu-8796.121.3~7/RELEASE_ARM64_T8103
    Binaries:
      Node: 16.20.0
      npm: 8.19.4
      Yarn: 1.22.19
      pnpm: 8.6.0
    Relevant Packages:
      next: 13.4.11
      eslint-config-next: 13.4.11
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
    Next.js Config:
      output: N/A
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

Image optimization (next/image, next/legacy/image)

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/gauravsaini964/image-preload-next13

### To Reproduce

1. Build the repo.
2. Navigate to index page (which is using app-dir) and inspect  index.html code.
3. Image preload statement is after font,css,webpack and 2-3 other chunks.
4. Navigate to /page-dir (which is using page dir) and repeat step 2 & 3.
5. Image preload is before all other preload tags.

### Describe the Bug

Order of image preload tag in app-dir is different from what it used to be in pages-dir and hence FP and LCP metric is getting affected. Same is replicated in lighthouse scores.
Compare following screenshots:

<img width="1870" alt="Screenshot 2023-07-21 at 4 21 41 PM" src="https://github.com/vercel/next.js/assets/16620277/084d42ff-9969-4859-bfc0-d640f790d34d">

<img width="2066" alt="Screenshot 2023-07-21 at 4 23 43 PM" src="https://github.com/vercel/next.js/assets/16620277/65eeaf1d-9a68-4a3a-b522-8b8998dc36a7">

<img width="1253" alt="Screenshot 2023-07-21 at 4 34 31 PM" src="https://github.com/vercel/next.js/assets/16620277/d33eb057-4643-440a-adf7-d7e655dd2af9">

<img width="1074" alt="Screenshot 2023-07-21 at 4 30 13 PM" src="https://github.com/vercel/next.js/assets/16620277/55e0924e-30a3-47ce-abee-6c39c60a35d0">

<img width="1222" alt="Screenshot 2023-07-21 at 4 33 29 PM" src="https://github.com/vercel/next.js/assets/16620277/4c911b35-b7a5-4eeb-bed2-6e0495c882b3">

<img width="998" alt="Screenshot 2023-07-21 at 4 31 05 PM" src="https://github.com/vercel/next.js/assets/16620277/89446e9d-f7f8-49fb-9686-e6a9a0099a3c">

### Expected Behavior

Image preload tag should come above every other preload tag imo as it is giving me best performance. 

I have a ecommerce app which has 1800+ pages built with both app-router and page-router. App router app is faster in every aspect apart from FP and LCP metric consistently by ~1seconds because that website has lot more scripts to preload.

### Which browser are you using? (if relevant)

Chrome

### How are you deploying your application? (if relevant)

Vercel

<sub>[NEXT-1463](https://linear.app/vercel/issue/NEXT-1463/nextimage-preload-not-working-as-expected-in-appdir-affecting-lcp-and)</sub>

**Comment by gauravsaini964** [comment]
Any update or ETA on this?
Sitting on website that needs to go live. 

Thanks for your efforts.

**Comment by styfle** [comment]
This is an issue in React.

I notified @gnoff who is going to fix it upstream.

**Comment by gauravsaini964** [comment]
Hey @gnoff. Any issue, PR or ETA to track?

Thank you for your contributions.

**Comment by gauravsaini964** [comment]
Hi @styfle & @gnoff
Please share if you have any update. Sorry for rushing. One of the last few bugs pending for our new website rollout.

**Comment by gnoff** [comment]
No issue just yet but I think a general timeline of 2 weeks is reasonable. I'll try to update this issue with a link tracking the work once I get it started

**Comment by gauravsaini964** [comment]
Hey @gnoff
Any progress on this? Thanks

**Comment by styfle** [comment]
I believe this is the upstream React PR: 

- https://github.com/facebook/react/pull/27191

**Comment by steve-marmalade** [comment]
Do we expect that this issue has been fixed since the upstream react fixes were included in `13.4.15` via https://github.com/vercel/next.js/pull/53881 ?

**Comment by styfle** [comment]
Confirmed this is fixed 👍
